packages/assets-controllers/src/AssetsContractControllerWithNetworkClientId.test.ts

```diff
@@ -0,0 +1,843 @@
+        ERC20_UNI_ADDRESS,
+        'invalidNetworkClientId',
+      ),
+    ).rejects.toThrow(expect.any(Error));
+    messenger.clearEventSubscriptions('NetworkController:stateChange');
+  });
+
+  it('should get balance of ERC-20 token contract correctly', async () => {
+    const { assetsContract, messenger, networkClientConfiguration } =
+      await setupControllers();
+    mockNetworkWithDefaultChainId({
+      networkClientConfiguration,
+      mocks: [
+        {
+          request: {
+            method: 'eth_call',
+            params: [
+              {
+                to: ERC20_UNI_ADDRESS,
+                data: '0x70a082310000000000000000000000005a3ca5cd63807ce5e4d7841ab32ce6b6d9bbba2d',
+              },
+              'latest',
+            ],
+          },
+          response: {
+            result:
+              '0x0000000000000000000000000000000000000000000000001765caf344a06d0a',
+          },
+        },
+        {
+          request: {
+            method: 'eth_call',
+            params: [
+              {
+                to: ERC20_UNI_ADDRESS,
+                data: '0x70a08231000000000000000000000000202637daaefbd7f131f90338a4a6c69f6cd5ce91',
+              },
+              'latest',
+            ],
+          },
+          response: {
+            result:
+              '0x0000000000000000000000000000000000000000000000000000000000000000',
+          },
+        },
+      ],
+    });
+    const UNIBalance = await assetsContract.getERC20BalanceOf(
+      ERC20_UNI_ADDRESS,
+      TEST_ACCOUNT_PUBLIC_ADDRESS,
+      'mainnet',
+    );
+    const UNINoBalance = await assetsContract.getERC20BalanceOf(
+      ERC20_UNI_ADDRESS,
+      '0x202637dAAEfbd7f131f90338a4A6c69F6Cd5CE91',
+      'mainnet',
+    );
+    expect(UNIBalance.toString(16)).not.toBe('0');
+    expect(UNINoBalance.toString(16)).toBe('0');
+    messenger.clearEventSubscriptions('NetworkController:stateChange');
+  });
+
+  it('should get ERC-721 NFT tokenId correctly', async () => {
+    const { assetsContract, messenger, networkClientConfiguration } =
+      await setupControllers();
+    mockNetworkWithDefaultChainId({
+      networkClientConfiguration,
+      mocks: [
+        {
+          request: {
+            method: 'eth_call',
+            params: [
+              {
+                to: ERC721_GODS_ADDRESS,
+                data: '0x2f745c590000000000000000000000009a90bd8d1149a88b42a99cf62215ad955d6f498a0000000000000000000000000000000000000000000000000000000000000000',
+              },
+              'latest',
+            ],
+          },
+          response: {
+            result:
+              '0x00000000000000000000000000000000000000000000000000000000000025a2',
+          },
+        },
+      ],
+    });
+    const tokenId = await assetsContract.getERC721NftTokenId(
+      ERC721_GODS_ADDRESS,
+      '0x9a90bd8d1149a88b42a99cf62215ad955d6f498a',
+      0,
+      'mainnet',
+    );
+    expect(tokenId).not.toBe(0);
```

Comment:
I see that these tests are copied over but ugh why are we testing this negative assertion... seems really brittle and prone to false positives...

Reply:
yeah, it's an odd test. I've been trying to avoid addressing unrelated tech debt with all our changes unless absolutely necessary. I think we leave this as is